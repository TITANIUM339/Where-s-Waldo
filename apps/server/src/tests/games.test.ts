import express from "express";
import request from "supertest";
import { describe, expect, test } from "vitest";
import errorHandler from "../middleware/error-handler";
import gamesRouter from "../routes/games/games";

const app = express();

app.use(gamesRouter);

app.use(errorHandler);

describe("responds with all games", () => {
    test("GET /games", async () => {
        const response = await request(app).get("/games");

        expect(response.status).toBe(200);
        expect(response.body).toMatchSnapshot();
    });
});
