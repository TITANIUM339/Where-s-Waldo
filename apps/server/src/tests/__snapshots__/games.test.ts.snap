// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`responds with all games > GET /games 1`] = `
[
  {
    "description": "WHAT A SPECTACULAR SCENE THIS IS, WALDO-WATCHERS! AND WHAT AN EPIC COMMOTION PICTURE! I WONDER WHY THE TROJANS DIDN'T GUESS THE WOODEN HORSE WAS FULL OF GREEKS, AND HOW DID THEY GET IT THROUGH THE GATES OF TROY ANYWAY? I WOULDN'T LIKE TO BE IN THE TROJANS' SANDALS, IF THE COSTUME DEPARTMENT HAD GIVEN THEM ANY, THAT IS!",
    "id": 1,
    "imageHeight": 2227,
    "imageURL": "https://i.imgur.com/QLACwOG.jpeg",
    "imageWidth": 3536,
    "name": "HORSEPLAY IN TROY",
  },
  {
    "description": "PHEW, INCREDIBLE! TIME, SPACE, AND HORROR ARE IN A MIGHTY MUDDLE HERE! WHAT COSMIC COSTUMES AND WHAT GREAT SPECIAL EFFECTS! ONE OF THOSE FLYING SAUCERS LOOKS LIKE IT'S REALLY FLYING! ARE THOSE REAL ALIENS INSIDE, NOT ACTORS AT ALL? SO WHAT'S REAL AND WHAT'S MADE UP IN FILMS LIKE THESE?",
    "id": 2,
    "imageHeight": 2223,
    "imageURL": "https://i.imgur.com/wg0Npy8.jpeg",
    "imageWidth": 3538,
    "name": "DINOSAURS, SPACEMEN, AND GHOULS",
  },
  {
    "description": "WHAT A CRUSH IN THE CAVE, WALDO-FOLLOWERS, BUT PAN IN ON THOSE POTS OF TREASURE! HOW MANY THIEVES WERE IN THE STORY? I BELIEVE THIS DIRECTOR THINKS FORTY THOUSAND! HAVE YOU SPOTTED ALI BABA? HE'S IN THE ALLEY, CUTTING HAIR - THE SCRIPTWRITER THINKS HIS NAME'S ALLEY BARBER! JANGLING GENIES - WHAT A FEARFULLY FUNNY FLICK THIS IS!",
    "id": 3,
    "imageHeight": 2226,
    "imageURL": "https://i.imgur.com/91qfVY7.jpeg",
    "imageWidth": 3505,
    "name": "ALI BABA AND THE FORTY THIEVES",
  },
  {
    "description": "THEN WALDO AND WIZARD WHITEBEARD CAME TO THE LAND OF THE UNFRIENDLY GIANTS. WHERE MANY WALDOS HAD BEEN BEFORE. AND WALDO SAW THAT THE GIANTS WERE HORRIDLY HARASSING THE LITTLE PEOPLE. AND WHEN HE FOUND THE TENTH SCROLL, IT WAS TIME TO CONTINUE WITH HIS JOURNEY.",
    "id": 4,
    "imageHeight": 2256,
    "imageURL": "https://i.imgur.com/OtAW3FQ.jpeg",
    "imageWidth": 3564,
    "name": "THE UNFRIENDLY GIANTS",
  },
]
`;
